OrbitQt/orbitsamplingreport.cpp

```diff
@@ -94,13 +94,10 @@ void OrbitSamplingReport::OnCurrentThreadTabChanged(int current_tab_index) {
     return;
   }
 
-  auto treeView = m_OrbitDataViews.at(current_tab_index);
+  auto treeView = m_OrbitDataViews[current_tab_index];
   QModelIndexList index_list = treeView->GetTreeView()->selectionModel()->selectedIndexes();
-  if (!index_list.isEmpty() && index_list.front().isValid()) {
-    treeView->GetTreeView()->GetModel()->OnRowSelected(index_list.front().row());
-  } else {
-    m_SamplingReport->ClearReport();
-  }
+  int row = (!index_list.isEmpty() && index_list.front().isValid()) ? index_list.front().row() : -1;
```

Comment:
I don't like OnRowSelected being executed with this -1.
I was thinking that we can use std::optional here. So, you give an optional row to OnRowSelected and you don't need this -1 constant here, neither in "OnSelect" function.

Also, if you want, function_address could be a std::optional.

Reply:
Thanks so much for bringing that to my eyes! I like std::optional!